File: packages/tasks-gen/snippets-fixtures/image-to-image/js/huggingface.js/0.fal-ai.js
import { InferenceClient } from "@huggingface/inference";

const client = new InferenceClient(process.env.HF_TOKEN);

const data = fs.readFileSync("cat.png");

const image = await client.imageToImage({
	provider: "fal-ai",
	model: "black-forest-labs/FLUX.1-Kontext-dev",
	inputs: data,
	parameters: { prompt: "Turn the cat into a tiger.", },
});
/// Use the generated image (it's a Blob)
// For example, you can save it to a file or display it in an image element

File: packages/tasks-gen/snippets-fixtures/image-to-image/python/huggingface_hub/0.fal-ai.py
from huggingface_hub import InferenceClient, get_token

client = InferenceClient(
    provider="fal-ai",
    api_key=get_token(),
)

with open("cat.png", "rb") as image_file:
   input_image = image_file.read()

# output is a PIL.Image object
image = client.image_to_image(
    input_image,
    prompt="Turn the cat into a tiger.",
    model="black-forest-labs/FLUX.1-Kontext-dev",
)